server/start.go

```diff
@@ -109,30 +109,30 @@ const (
 )
 
 // StartCmdOptions defines options that can be customized in `StartCmdWithOptions`,
-type StartCmdOptions struct {
+type StartCmdOptions[T types.Application] struct {
 	// DBOpener can be used to customize db opening, for example customize db options or support different db backends,
 	// default to the builtin db opener.
 	DBOpener func(rootDir string, backendType dbm.BackendType) (dbm.DB, error)
 	// PostSetup can be used to setup extra services under the same cancellable context,
 	// it's not called in stand-alone mode, only for in-process mode.
-	PostSetup func(svrCtx *Context, clientCtx client.Context, ctx context.Context, g *errgroup.Group) error
+	PostSetup func(app T, svrCtx *Context, clientCtx client.Context, ctx context.Context, g *errgroup.Group) error
 	// PostSetupStandalone can be used to setup extra services under the same cancellable context,
-	PostSetupStandalone func(svrCtx *Context, clientCtx client.Context, ctx context.Context, g *errgroup.Group) error
+	PostSetupStandalone func(app T, svrCtx *Context, clientCtx client.Context, ctx context.Context, g *errgroup.Group) error
 	// AddFlags add custom flags to start cmd
 	AddFlags func(cmd *cobra.Command)
 	// StartCommandHanlder can be used to customize the start command handler
-	StartCommandHandler func(svrCtx *Context, clientCtx client.Context, appCreator types.AppCreator, inProcessConsensus bool, opts StartCmdOptions) error
+	StartCommandHandler func(svrCtx *Context, clientCtx client.Context, appCreator types.AppCreator[T], inProcessConsensus bool, opts StartCmdOptions[T]) error
```

Comment:
can you explain what the benefit is for app devs with this? i saw you mentioned berachain, but curious what they/others use it for.

Reply:
Beside allowing `PostSetup` to access and manipulate `app` instance directly in the setup process. It wasn't  answer in the original PR.  Look like a hook that allows adding custom logic after the setup is complete but before the app starts running. Like initializing additional components, setting up background processes (data metrics collect ??), configuring middleware..

Reply:
I'm not sure 🤔 also I can't tag him